kernel/src/schema/mod.rs

```diff
@@ -212,33 +216,107 @@ impl StructField {
             .collect()
     }
 
-    /// Applies physical name mappings to this field
+    /// Applies physical name mappings to this field. If the `column_mapping_mode` is
+    /// [`ColumnMappingMode::Id`], then each StructField will have its parquet field id in the
+    /// [`ColumnMetadataKey::ParquetFieldId`] metadata field.
```

Comment:
can we add more explicit docs about what is done here? I think would be useful to call out that we add field_id only if id mode is set and we remove it if not (right?) - then readers should use the field_id presence to determine how to resolve columns (present = id resolution, missing = name resolution)

Reply:
Updated the docs. PTAL